test/calibration/test_calibrations.py

```diff
@@ -0,0 +1,513 @@
+
+        self.sigma = Parameter("σ")
+        self.amp_xp = Parameter("amp")
+        self.amp = Parameter("amp")
+        self.beta = Parameter("β")
+        self.drive = DriveChannel(Parameter("ch0"))
+        self.date_time = datetime.strptime("15/09/19 10:21:35", "%d/%m/%y %H:%M:%S")
+
+        # Template schedule for qubit 3
+        with pulse.build(name="xp") as xp_drag:
+            pulse.play(Drag(160, self.amp_xp, self.sigma, self.beta), self.drive)
+
+        # Default template schedule for all qubits
+        with pulse.build(name="xp") as xp:
+            pulse.play(Gaussian(160, self.amp, self.sigma), self.drive)
+
+        # Add the schedules
+        self.cals.add_schedule(xp)
+        self.cals.add_schedule(xp_drag, (3,))
+
+    def test_parameter_value_adding_and_filtering(self):
+        """Test that adding parameter values behaves in the expected way."""
+
+        # Ensure that no parameter values are present when none have been added.
+        params = self.cals.parameters_table()
+        self.assertEqual(params, [])
+
+        # Add a default parameter common to all qubits.
+        self.cals.add_parameter_value(ParameterValue(40, self.date_time), "σ", None, "xp")
+        self.assertEqual(len(self.cals.parameters_table()), 1)
+
+        # Check that we can get a default parameter in the parameter table
+        self.assertEqual(len(self.cals.parameters_table(parameters=["σ"])), 1)
+        self.assertEqual(len(self.cals.parameters_table(parameters=["σ"], schedules=["xp"])), 1)
+        self.assertEqual(len(self.cals.parameters_table(parameters=["σ"], schedules=["xm"])), 0)
+
+        # Test behaviour of qubit-specific parameter and without ParameterValue.
+        self.cals.add_parameter_value(0.25, "amp", (3,), "xp")
+        self.cals.add_parameter_value(0.15, "amp", (0,), "xp")
+
+        # Check the value for qubit 0
+        params = self.cals.parameters_table(parameters=["amp"], qubit_list=[(0,)])
+        self.assertEqual(len(params), 1)
+        self.assertEqual(params[0]["value"], 0.15)
+        self.assertEqual(params[0]["qubits"], (0,))
+
+        # Check the value for qubit 3
+        params = self.cals.parameters_table(parameters=["amp"], qubit_list=[(3,)])
+        self.assertEqual(len(params), 1)
+        self.assertEqual(params[0]["value"], 0.25)
+        self.assertEqual(params[0]["qubits"], (3,))
+
+    def _add_parameters(self):
+        """Helper function."""
+
+        # Add the minimum number of parameter values. Sigma is shared across both schedules.
+        self.cals.add_parameter_value(ParameterValue(40, self.date_time), "σ", schedule="xp")
+        self.cals.add_parameter_value(ParameterValue(0.25, self.date_time), "amp", (3,), "xp")
+        self.cals.add_parameter_value(ParameterValue(0.15, self.date_time), "amp", (0,), "xp")
+        self.cals.add_parameter_value(ParameterValue(10, self.date_time), "β", (3,), "xp")
+
+    def test_default_schedules(self):
```

Comment:
The way Schedules, Parameters and ParameterExpressions  deal with copy and assignment is a little funny. It might be good to add a check that the original parametrized schedule is unmodified (though I think this test implies it).

Reply:
Agree. See here: bc11c38